lib/mongo/socket/ssl.rb

```diff
@@ -122,23 +124,18 @@ def create_context(options)
       end
 
       def set_cert(context, options)
-        context.cert = OpenSSL::X509::Certificate.new(File.open(options[:ssl_cert]))
+        context.cert = certificate_from_option(options[:ssl_cert])
       end
 
       def set_key(context, options)
-        if options[:ssl_key_pass_phrase]
-          context.key = OpenSSL::PKey::RSA.new(File.open(options[:ssl_key]),
-                                               options[:ssl_key_pass_phrase])
-        else
-          context.key = OpenSSL::PKey::RSA.new(File.open(options[:ssl_key]))
-        end
+        context.key = key_from_option(options[:ssl_key], options[:ssl_key_pass_phrase])
```

Comment:
the options for this would be:
``ssl_key`` : file path
``ssl_key_string`` or ``ssl_key_raw`` : string
``ssl_key_obj`` or ``ssl_key_object`` for the OpenSSL::PKey::RSA

``ssl_key_pass_phrase`` stays as-is